store/postgres/src/notification_listener.rs

```diff
@@ -158,27 +192,28 @@ impl NotificationListener {
                     //
                     // We batch notifications such that we do not wait for
                     // longer than 500ms for new notifications to arrive,
-                    // but limit the size of each batch to 64 to guarantee
+                    // but limit the size of each batch to 128 to guarantee
                     // progress on a busy system
                     let notifications: Vec<_> = conn
                         .notifications()
                         .timeout_iter(Duration::from_millis(500))
                         .iterator()
+                        .take(128)
                         .filter_map(|item| match item {
                             Ok(msg) => Some(msg),
                             Err(e) => {
-                                let msg = format!("{}", e);
-                                crit!(logger, "Error receiving message"; "error" => &msg);
-                                eprintln!(
-                                    "Connection to Postgres lost while listening for events. \
-                             Aborting to avoid inconsistent state. ({})",
-                                    msg
-                                );
-                                std::process::abort();
+                                // When there's an error, process whatever
+                                // notifications we've picked up so far, and
+                                // cause the start of the loop to reconnect
+                                if connected {
+                                    let msg = format!("{}", e);
+                                    crit!(logger, "Error receiving message"; "error" => &msg);
+                                }
+                                connected = false;
+                                None
                             }
                         })
                         .filter(|notification| notification.channel() == channel_name.0)
-                        .take(64)
```

Comment:
No, it's fairly random; we just can't wait until we got all messages in the queue since we might never finish getting messages on a busy system. That number is just made up, but seems to work fine.